test/e2e/multihoming.go

```diff
@@ -807,6 +829,113 @@ var _ = Describe("Multi Homing", func() {
 				})
 
+				Context("and networkAttachmentDefinition is modified", func() {
+					const (
+						expectedChangedMTU        = 1600
+						newDesiredRange           = "60.128.0.192/28" // Desired IPs from 60.128.0.192 to 60.128.0.207
+						excludedSubnetLowerRange1 = "60.128.0.0/25"   // Excludes IPs from 60.128.0.0 to 60.128.0.127
+						excludedSubnetLowerRange2 = "60.128.0.128/26" // Excludes IPs from 60.128.0.128 to 60.128.0.191
+						excludedSubnetUpperRange1 = "60.128.0.208/28" // Excludes IPs from 60.128.0.208 to 60.128.0.223
+						excludedSubnetUpperRange2 = "60.128.0.224/27" // Excludes IPs from 60.128.0.224 to 60.128.0.255
+						newLocalnetVLANID         = 30
+					)
+					BeforeEach(func() {
+						By("setting new MTU")
+						netConfig.mtu = expectedChangedMTU
+						By("setting new subnets to leave a smaller range")
+						netConfig.excludeCIDRs = []string{excludedSubnetLowerRange1, excludedSubnetLowerRange2, excludedSubnetUpperRange1, excludedSubnetUpperRange2}
+						By("setting new VLAN-ID")
+						netConfig.vlanID = newLocalnetVLANID
+						p := []byte(fmt.Sprintf(`[{"op":"replace","path":"/spec/config","value":%q}]`, generateNADSpec(netConfig)))
+						Expect(patchNADSpec(nadClient, netConfig.name, netConfig.namespace, p)).To(Succeed())
+					})
+
+					It("sets the new MTU on the pod after NetworkAttachmentDefinition reconcile", func() {
+						Eventually(func(g Gomega) error {
+							clientPodConfig := podConfiguration{
+								name:        clientPodName + randStr(10),
+								namespace:   f.Namespace.Name,
+								attachments: []nadapi.NetworkSelectionElement{{Name: secondaryNetworkName}},
+							}
+							kickstartPod(cs, clientPodConfig)
+							mtu, err := getSecondaryInterfaceMTU(clientPodConfig)
+							if err != nil {
+								return fmt.Errorf("failed to get MTU: %w", err)
+							}
+							if mtu != expectedChangedMTU {
+								g.Expect(cs.CoreV1().Pods(clientPodConfig.namespace).Delete(context.Background(), clientPodConfig.name, metav1.DeleteOptions{})).To(Succeed())
+								return fmt.Errorf("pod MTU is %d, but expected %d", mtu, expectedChangedMTU)
+							}
+							return nil
+						}).Should(Succeed(), "pod MTU should be properly configured")
+					})
+
+					It("allocates the pod's secondary interface IP in the new range after NetworkAttachmentDefinition reconcile", func() {
+						By("asserting the pod's secondary interface IP is properly configured")
+						Eventually(func(g Gomega) error {
+							clientPodConfig := podConfiguration{
+								name:        clientPodName + "-" + randStr(10),
+								namespace:   f.Namespace.Name,
+								attachments: []nadapi.NetworkSelectionElement{{Name: secondaryNetworkName}},
+							}
+							kickstartPod(cs, clientPodConfig)
+
+							clientIP, err := podIPForAttachment(cs, clientPodConfig.namespace, clientPodConfig.name, netConfig.name, 0)
+							g.Expect(err).NotTo(HaveOccurred())
```

Comment:
This is a nit and you may ignore it, but ....

If you are going to be combining returning err with g.Expect inside the Eventually block (which is fine but sets us up a bit for ambiguity, I was originally giving you the two options expecting you to pick one or the other instead of combining them) then I would return error for the main business logic of the Eventually block and use Expect just on side-effects. 

With that I am saying that I would return this error instead of Expecting on it.
Make sense to interpret pod delete as a side effect though.